### PR TITLE
Do not throw an exception on existing field access not returned by th…

### LINKED ISSUE #130: Do not throw an exception on existing field access not returned by the api
Accessing to a field can throw exception like `"actions is not a field of FacebookAds\\Object\\Insights"`.

This is wrong: actions is a field of Insights, it's accessible for several objects, but, when the field is not returned by facebook api, the sdk throw this exception.

The issue is in [AbstractObject](https://github.com/facebook/facebook-php-ads-sdk/blob/master/src/FacebookAds/Object/AbstractObject.php#L49). The presence of the field in the enum should be asserted if the api result is null before throwing an exception.

And this leads to ugly catch in client code:
```php
try {
  $actions = $entity->{InsightsFields::ACTIONS};
} catch(\InvalidArgumentException $e) {
  $actions = null;
};
```

### PR DESCRIPTION
**Pull request by yamsellem**
…e api

fixes #130

### CODE REVIEW

File: src/FacebookAds/Object/AbstractObject.php [hunk 1]
```diff
@@ -45,6 +45,8 @@ public function __construct() {
   public function __get($name) {
     if (array_key_exists($name, $this->data)) {
       return $this->data[$name];
+    } elseif (in_array($name, static::getFieldsEnum()->getValues())) {
```
Comment (pruno) [inline src/FacebookAds/Object/AbstractObject.php]: The logic looks good, but the enum already offer an index based validation function that will perform better than in_array.

can you replace

~~~php
in_array($name, static::getFieldsEnum()->getValues())
~~~

with

~~~php
static::getFieldsEnum()->isValid($name)
~~~

?

Comment (yamsellem) [inline src/FacebookAds/Object/AbstractObject.php]: It does not work.
It seems that `isValid` use the same keys as `getValuesMap` witch exclude field not returned by the api.

I've tried several ways but only found this O(n) solution.
Any other idea?

_ps. don't you have any method to try to reproduce this — maybe unit test, this should be more efficient than suggesting fix ;-)_

Comment (pruno) [inline src/FacebookAds/Object/AbstractObject.php]: I think i found  the issue, which is not actually here. but in AbstractObject::setData() that will override the $this->data structure.
https://github.com/facebook/facebook-php-ads-sdk/commit/134eca31ad46821f7d12efc743a87b4900296de9 should fix it. would you mind try it?

Comment (yamsellem) [inline src/FacebookAds/Object/AbstractObject.php]: @pruno you're right, [134eca3](https://github.com/facebook/facebook-php-ads-sdk/commit/134eca31ad46821f7d12efc743a87b4900296de9) fix this ;-) 
Great!

Any idea of a release date?

Comment (pruno) [inline src/FacebookAds/Object/AbstractObject.php]: not yet, but since this bug can be nasty i'll probably ship a new patch version soon

**Pull request closed without merging**